src/main/java/GameField.java

```diff
@@ -1,132 +1,35 @@
-        if (gameObject.isPermeable()) {
-            return true;
-        }
-
-        Point speed = gameObject.getSpeed();
-
-        Rectangle objectMovedBody = new Rectangle(gameObject.getBody());
-        objectMovedBody.shift(speed);
+        this.gameObjectManager = new GameObjectManager();
+        this.collisionManager = new CollisionManager(gameObjectManager);
 
-        if (isOutOfGameField(objectMovedBody)) {
-            return false;
-        }
-
-        if (doesIntersectsPermeableGameObjectsExcept(objectMovedBody, gameObject)) {
-            return false;
-        }
-
-        return true;
-    }
-
-    private void processCollisions() {
-        registerCollisions();
-        carryOutRegisteredCollisions();
-        clearRegisteredCollisions();
+        Rectangle field = new Rectangle(leftTop, rightBottom);
+        this.physicManager = new PhysicManager(field, gameObjectManager);
     }
 
-    private void registerCollisions() {
-        collisions.clear();
-        for (GameObject gameObject : gameObjects) {
-            registerCollision(gameObject);
+    void addObject(GameObject gameObject) throws FailedAddObjectException {
+        if (!physicManager.canAddObject(gameObject)) {
+            throw new FailedAddObjectException();
         }
-    }
 
-    private void carryOutRegisteredCollisions() {
-        for (Collision collision : collisions) {
-            collision.carryOut();
-        }
+        gameObjectManager.addObject(gameObject);
     }
 
-    private void clearRegisteredCollisions() {
-        collisions.clear();
-    }
-
-    private void registerCollision(GameObject gameObject) {
-        for (GameObject otherGameObject : gameObjects) {
-            if (otherGameObject.equals(gameObject)) {
-                continue;
-            }
-
-            if (otherGameObject.doesIntersect(gameObject.getBody())) {
-                collisions.add(new Collision(gameObject, otherGameObject));
-            }
-        }
-    }
-
-    private boolean doesIntersectsPermeableGameObjectsExcept(Rectangle rectangle, GameObject exception) {
-        Collection<GameObject> gameObjectsWithoutException = new HashSet<>(gameObjects);
-        gameObjectsWithoutException.remove(exception);
-
-        return doesIntersectsPermeableGameObjects(rectangle, gameObjectsWithoutException);
-    }
-
-    private boolean doesIntersectsPermeableGameObjects(Rectangle rectangle, Collection<GameObject> objectsToIntersect) {
-        for (GameObject gameObject : objectsToIntersect) {
-            if (gameObject.isPermeable()) {
-                continue;
-            }
-            if (gameObject.doesIntersect(rectangle)) {
-                return true;
-            }
-        }
-        return false;
-    }
-
-    private boolean isOutOfGameField(Rectangle objectBody) {
-        return !field.covers(objectBody);
+    void update() {
+        physicManager.move();
+        collisionManager.processCollisions();
```

Comment:
then function `move()` will have too many responsibility that aren't specified in the name. Maybe rename `processCollision()` to `processInteraction()`, because it's not a physical process, rather game event, like bullet damages a game object or game object takes a bonus. No interaction in physical sense

Reply:
I think this conversation is outdated and we decided that PhysicManager should collect a collection of collision events report about it